ml-agents-envs/mlagents_envs/environment.py

```diff
@@ -377,6 +383,18 @@ def get_steps(
         self._assert_behavior_exists(behavior_name)
         return self._env_state[behavior_name]
 
+    def _poll_process(self) -> None:
+        """
+        Check the status of the subprocess. If it has exited, raise a UnityEnvironmentException
+        :return: None
+        """
+        if not self._proc1:
+            return
+        poll_res = self._proc1.poll()
+        if poll_res is not None:
+            exc_msg = self._returncode_to_env_message(self._proc1.returncode)
+            raise UnityEnvironmentException(exc_msg)
```

Comment:
Not sure if we want to have separate exception messages for returncode 0 and non-0.